src/docs/development/platform-integration/platform-channels.md

```diff
@@ -185,32 +187,31 @@ interface that displays the battery state in a string,
 and a button for refreshing the value.
 
 <!-- skip -->
+<?code-excerpt "main.dart" title?>
 ```dart
-@override
-Widget build(BuildContext context) {
-  return Material(
-    child: Center(
-      child: Column(
-        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-        children: [
-          RaisedButton(
-            child: Text('Get Battery Level'),
-            onPressed: _getBatteryLevel,
-          ),
-          Text(_batteryLevel),
-        ],
+  @override
+  Widget build(BuildContext context) {
+    return Material(
+      child: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+          children: [
+            RaisedButton(
+              child: Text('Get Battery Level'),
+              onPressed: _getBatteryLevel,
+            ),
+            Text(_batteryLevel),
+          ],
+        ),
       ),
-    ),
-  );
-}
+    );
+  }
 ```
 
+### Step 3: Add an Android platform-specific implementation
 
-### Step 3a: Add an Android platform-specific implementation using Java {#example-java}
-
-*Note*: The following steps use Java. If you prefer Kotlin, skip to step
-3b.
-
+{% samplecode android-channel %}
+{% sample Java %}
 Start by opening the Android host portion of your Flutter app in Android Studio:
```

Comment:
It looks like you're including plain text in a code block. Is that what you want?